src/pudl/transform/classes.py

```diff
@@ -883,18 +885,30 @@ def spot_fix_values(self, df: pd.DataFrame, params: SpotFixes) -> pd.DataFrame:
     Returns:
         The same input DataFrame but with some spot fixes corrected.
     """
-    # The default SpotFixValues() instance has no effect:
-    if params.record_id is None or params.record_col is None:
-        return df
+    spot_fixes_df = pd.DataFrame(
+        params.spot_fixes, columns=params.idx_cols + params.fix_cols
+    )
 
-    elif params.record_col in list(df.columns):  # If record_col in df
-        for key in params.fixes:  # For each fix
-            if key in list(df.columns):
-                df.loc[df[params.record_col] == params.record_id, key] = params.fixes[
-                    key
-                ]  # Manually update value
+    if not (spot_fixes_df.dtypes == df[params.idx_cols + params.fix_cols].dtypes).all():
+        raise ValueError(
+            "Spot fix datatypes do not match existing dataframe datatypes."
+        )
+    """Check that the datatypes of the spot fixed values match the existing data types."""
```

Comment:
I guess this is syntactically fine, but we've typically only used the trailing triple-quotes annotation on class attributes and constants, so that they'll show up in the autogenerated documentation, and use leading-# comments for inline annotation of our code.